packages/core/admin/admin/src/content-manager/history/pages/History.tsx

```diff
@@ -100,33 +100,31 @@ const HistoryPage = () => {
     }
   }, [versionsResponse.isLoading, navigate, query.id, versionsResponse.data?.data, query]);
 
-  const { permissions, isLoading: isLoadingPermissions } = useSyncRbac(slug!, query, 'History');
-
-  if (isLoadingDocument || isLoadingLayout || versionsResponse.isLoading || isLoadingPermissions) {
-    return <Page.Loading />;
-  }
-
   /**
    * Ensure that we have the necessary data to render the page:
    * - slug for single types
    * - slug _and_ documentId for collection types
    */
-  if (!slug || (!documentId && collectionType === COLLECTION_TYPES)) {
+  if (!slug || (collectionType === COLLECTION_TYPES && !documentId)) {
     return <Navigate to="/content-manager" />;
   }
 
-  // TODO: real error state when designs are ready
-  if (versionsResponse.isError || !versionsResponse.data || !layout || !schema || !permissions) {
-    return null;
+  if (isLoadingDocument || isLoadingLayout || versionsResponse.isLoading || !selectedVersionId) {
+    return <Page.Loading />;
+  }
```

Comment:
We forgot the case where you have no history versions: it means no selectedVersionId, so it gets stuck on this loader. You may need to return Loading in 2 different places